### PR TITLE
Fix CI for `mimoo/switch`

### PR DESCRIPTION
**Pull request by A-Manning**

### CODE REVIEW

File: scripts/compare_ci_diff_binables.sh [hunk 1]
```diff
@@ -9,10 +9,11 @@ fi
 
 # cleanup if needed
 
-git clean -dfx
+git clean -dfx -e _opam/
```
Comment (mimoo) [inline scripts/compare_ci_diff_binables.sh:12]: isn't adding `_opam` to .gitignore enough?

File: scripts/opam-env.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+eval $(opam env)
```
Comment (mimoo) [inline scripts/opam-env.sh:2]: this file doesn't seem to be called from anywhere

File: buildkite/src/Command/OpamInit.dhall [hunk 3]
```diff
@@ -23,6 +26,16 @@ let commands : List Text -> List Cmd.Type = \(environment : List Text) ->
...
+      },
+    Cmd.cacheThrough
+      Cmd.Docker::{
+        image = (../Constants/ContainerImages.dhall).minaToolchain,
+        extraEnv = environment
+      }
+      local
+      Cmd.CacheSetupCmd::{
+        create = r "[ -d _opam ] && : || make setup-opam",
+        package = r "tar cfz ${local} _opam"
```
Comment (mimoo) [inline buildkite/src/Command/OpamInit.dhall:38]: what's happening in this file exactly :o ?

**Comment by mimoo** [comment]
This is awesome! Thanks for pushing for this to work :) I left you some comments, feel free to merge!

File: scripts/compare_ci_diff_binables.sh (hunk 1 above)
Comment (A-Manning) [reply scripts/compare_ci_diff_binables.sh:12]: [`git clean` does not use standard ignore rules when the `-x` switch is set](https://git-scm.com/docs/git-clean#Documentation/git-clean.txt--x).

File: scripts/opam-env.sh (hunk 2 above)
Comment (A-Manning) [reply scripts/opam-env.sh]: Removed the file.

File: buildkite/src/Command/OpamInit.dhall (hunk 3 above)
Comment (A-Manning) [reply buildkite/src/Command/OpamInit.dhall:38]: When setting up opam, both the system and local switches need to be cached on the CI.

I've modified the file to ensure that the local opam switch is generated, (un)tarballed, and cached on the CI as necessary.

**Review by mimoo — APPROVED** [review]

**Pull request merged**